### PR TITLE
SRIOV provider: Deploy default CNI

### PR DESCRIPTION
**Pull request by ormergi**
While back then we decided to deploy Calico CNI on KinD providers
in order to create dual-stack cluster for DEV and CI env's.

As for today we managed to enable dual-stack clusters using our
VM based providers, and eventually removed the IPV6 KinD based provider.
Though we still deploy old version of Calico on the SRIOV lane provider.

We should use the default KinD CNI 'kindnet' instead,
it will reduce complexity of the setup and maintenance.

Signed-off-by: Or Mergi <ormergi@redhat.com>

### CODE REVIEW

**Comment by ormergi** [comment]
SRIOV lane with kindnet CNI plugin pass 
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/1462/rehearsal-pull-kubevirt-e2e-kind-1.19-sriov/1417481998527631360

**Comment by ormergi** [comment]
@phoracek @qinqon @maiqueb @EdDev SRIOV lane with kindnet CNI plugin pass 
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/1462/rehearsal-pull-kubevirt-e2e-kind-1.19-sriov/1417481998527631360

File: cluster-up/cluster/kind/common.sh [hunk 1]
```diff
@@ -102,7 +102,6 @@ function _configure_registry_on_node() {
 
 function _install_cnis {
     _install_cni_plugins
-    _install_calico_cni
```
Comment (EdDev) [inline cluster-up/cluster/kind/common.sh:105]: Can the function be deleted now?

**Review by EdDev — COMMENTED** [review]
Very nice!

File: cluster-up/cluster/kind/common.sh (hunk 1 above)
Comment (ormergi) [reply cluster-up/cluster/kind/common.sh:105]: Yes we can remove all related code, if we would want to use Calico again we have git history

Comment (ormergi) [reply cluster-up/cluster/kind/common.sh:105]: Done

**Comment by ormergi** [comment]
@EdDev I addressed you review comment could you please take another look?

**Review by EdDev — COMMENTED** [review]
Does this change mean we do not deploy dual-stack anymore on the SR-IOV lane? I could not understand that part from the commit message.

And if not, does `kindnet` support it (so we can add it)?

**Comment by ormergi** [comment]
> Does this change mean we do not deploy dual-stack anymore on the SR-IOV lane? I could not understand that part from the commit message.

Yes its single stack now,
The test that checks connectivity between VM's though SRIOV interface with IPv6 is green.

> And if not, does `kindnet` support it (so we can add it)?

In general kindnet supports dual-stack

**Comment by EdDev** [comment]
> In general kindnet supports dual-stack

If it is easily addable, then can we do it now? It is not a must, but it will be nicer so we will not need to think about it in the future.

**Comment by ormergi** [comment]
> > In general kindnet supports dual-stack
> 
> If it is easily addable, then can we do it now? It is not a must, but it will be nicer so we will not need to think about it in the future.

With the new Kind API its pretty straight forward https://kind.sigs.k8s.io/docs/user/configuration/#networking
the problem is that it supported on k8s-1.20+ clusters.
Due to the fact we still use `PodPreset` (deprecated on k8s-1.20) we cant upgrade the provider now.
As soon as https://github.com/kubernetes-sigs/kind/pull/2321 will get in we will be able to upgrade the provider 
stop using `PodPreset` and eventually turn on dual-stack.

**Review by EdDev — APPROVED** [review]
Thanks!

**Review by EdDev — CHANGES_REQUESTED** [review]
ohh, what about `./cluster-up/cluster/kind/manifests/kube-calico.yaml.in`?

EDIT: Also `./cluster-up/cluster/kind/manifests/kube-calico.diff.in`

**Comment by ormergi** [comment]
> ohh, what about `./cluster-up/cluster/kind/manifests/kube-calico.yaml.in`?
> 
> EDIT: Also `./cluster-up/cluster/kind/manifests/kube-calico.diff.in`

@EdDev Done

**Review by EdDev — APPROVED** [review]
Thanks!

**Review by enp0s3 — APPROVED** [review]
@ormergi Thank you!
/approve

**Pull request merged**